src/Windows.Core/SIZE_T.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// The maximum number of bytes to which a pointer can point.
+    /// Use for a count that must span the full range of a pointer.
+    /// </summary>
+    /// <remarks>
+    /// This struct is intended for P/Invoke marshaling and must exactly match
+    /// the size of <see cref="IntPtr"/>
+    /// </remarks>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct SIZE_T
+    {
+        private IntPtr size;
```

Comment:
Why not use `UIntPtr` instead, given `SIZE_T` is an unsigned integer? That would eliminate some casts you have to use below.